cmd/object-api-datatypes.go

```diff
@@ -178,6 +179,22 @@ type ObjectInfo struct {
 	SuccessorModTime time.Time
 }
 
+// ArchiveInfo returns any saved zip archive meta information
+func (o ObjectInfo) ArchiveInfo() []byte {
+	if len(o.UserDefined) == 0 {
+		return nil
+	}
+	z, ok := o.UserDefined[archiveInfoMetadataKey]
+	if !ok {
+		return nil
+	}
+	zipInfo, err := base64.StdEncoding.DecodeString(z)
+	if err != nil {
+		zipInfo = []byte(z)
+	}
+	return zipInfo
```

Comment:
First byte will currently be 1,2 or 3, but may be higher in the future. Use that as a faster and safer check:

```suggestion
	// FS/gateway mode does base64 encoding on roundtrip.
	// zipindex has version as first byte, which is below any base64 value.
	if len(z) > 0 && z[0] >= 32 {
		if zipInfo, err := base64.StdEncoding.DecodeString(z); err == nil {
			return zipInfo
		}
	}
	return []byte(z)
```